demisto_sdk/commands/download/downloader.py

```diff
@@ -750,520 +863,434 @@ def get_playbook_id_by_playbook_name(self, playbook_name: str) -> Optional[str]:
+            output_path (Path): The output path to write the files to.
+
+        Returns:
+            bool: True if all files were downloaded successfully, False otherwise.
         """
-        for custom_content_object in self.custom_content:
-
-            file_entity: str = custom_content_object[
-                "entity"
-            ]  # For example: Integrations
-            exist_in_pack: bool = custom_content_object[
-                "exist_in_pack"
-            ]  # For example: True
-            file_name: str = custom_content_object["name"]  # For example: Hello World
-            file_ending: str = custom_content_object["file_ending"]  # For example: yml
-
-            if exist_in_pack:
-                if self.force:
-                    if file_entity in (INTEGRATIONS_DIR, SCRIPTS_DIR):
-                        self.merge_and_extract_existing_file(custom_content_object)
-                    else:
-                        self.merge_existing_file(custom_content_object, file_ending)
+        logger.info("Saving downloaded files to output path...")
+        successful_downloads_count = 0
+        existing_files_skipped_count = 0
+        failed_downloads_count = 0
+
+        for file_name, content_object in downloaded_content_objects.items():
+            content_item_name: str = content_object["name"]
+            content_item_entity: str = content_object["entity"]
+            content_item_not_skipped = True
+
+            try:
+                if content_item_entity in (INTEGRATIONS_DIR, SCRIPTS_DIR):
+                    file_downloaded = self.download_unified_content(content_object=content_object,
+                                                                    existing_pack_structure=existing_pack_structure,
+                                                                    output_path=output_path,
+                                                                    overwrite_existing=self.force)
+
                 else:
-                    self.files_not_downloaded.append([file_name, FILE_EXIST_REASON])
+                    file_downloaded = self.download_non_unified_content(content_object=content_object,
+                                                                        existing_pack_structure=existing_pack_structure,
+                                                                        output_path=output_path,
+                                                                        overwrite_existing=self.force)
+
+                # If even one file was skipped, we mark the file as skipped for the logs
+                if not file_downloaded:
+                    content_item_not_skipped = False
+
+            except Exception as e:
+                failed_downloads_count += 1
+                logger.error(f"Failed to download content item '{content_item_name}': {e}")
+                logger.debug(traceback.format_exc())
```

Comment:
Same as before... this will cause the traceback to appear in the CLI output for users, and I think it's more fitting to include it only in debug logs.